core/src/no/group15/playmagic/ecs/GameMap.kt

```diff
@@ -152,15 +152,16 @@ class GameMap (
 
 
 	private fun boundingBox (entity: Entity): Rectangle {
+		// TODO replace this with existing boundingbox in transform component
 		val transform = entity.getComponent(TransformComponent::class.java)
 		val texture = entity.getComponent(TextureComponent::class.java)
 		val coordinate = toMatrixIndexes(Coordinate(transform.position.x, transform.position.y))
 
 		return Rectangle(
 			coordinate.x,
 			coordinate.y,
-			texture.size.x * transform.scale.x,
-			texture.size.y * transform.scale.y
+			transform.boundingBox.width * transform.scale.x,
+			transform.boundingBox.width * transform.scale.y
```

Comment:
line 164 should be transform.boundingBox.height? Or are you assuming it is a square?
```suggestion
			transform.boundingBox.width * transform.scale.x,
			transform.boundingBox.height * transform.scale.y
```